azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement.go

```diff
@@ -0,0 +1,378 @@
+package memberentitlementmanagement
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/ahmetb/go-linq"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/graph"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/identity"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/licensing"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/memberentitlementmanagement"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/webapi"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/suppress"
+)
+
+var (
+	groupConfigurationKeys = []string{
+		"display_name",
+		"origin_id",
+	}
+)
+
+func ResourceGroupEntitlement() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGroupEntitlementCreate,
+		Read:   resourceGroupEntitlementRead,
+		Delete: resourceGroupEntitlementDelete,
+		Update: resourceGroupEntitlementUpdate,
+		Importer: &schema.ResourceImporter{
+			State: importGroupEntitlement,
+		},
+		Schema: map[string]*schema.Schema{
+			"principal_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```

Comment:
`principal_name`  cannot be updated
```suggestion
				Computed: true,
```